jupyterlab_git/handlers.py

```diff
@@ -37,6 +38,17 @@ class GitHandler(APIHandler):
     def git(self) -> Git:
         return self.settings["git"]
 
+    def prepare(self):
+        """Check if the path should be skipped"""
+        super().prepare()
+        path = self.path_kwargs.get("path")
+        if path is not None:
+            excluded_paths = self.git.excluded_paths
+            for excluded_path in excluded_paths:
+                if re.match(excluded_path, path):
+                    self.set_status(404)
+                    self.finish()
```

Comment:
We should use [fnmatchcase](https://docs.python.org/3/library/fnmatch.html) instead of regex as it is more commonly expected - as proven by your test `/test-path/*` in regex means `/test-path` followed by any number of `/`; not `/test-path/` followed by anything (as I suppose you meant).

```suggestion
            for excluded_path in excluded_paths:
                if fnmatch.fnmatchcase(path, excluded_path):
                    self.set_status(404)
                    self.finish()
                    break
```

Reply:
Got it, I was just looking into better ways of doing the pattern match.